### PR TITLE
[release/3.1] Move off of expiring machine pools

### PR DESCRIPTION
**Pull request by wtgodbe**
Part of https://github.com/dotnet/aspnetcore-internal/issues/3950

### CODE REVIEW

File: .azure/pipelines/ci.yml [hunk 1]
```diff
@@ -626,7 +626,7 @@ stages:
     parameters:
       condition: ne(variables['SkipTests'], 'true')
       jobName: MacOS_Test
-      jobDisplayName: "Test: macOS 10.14"
+      jobDisplayName: "Test: macOS 10.15"
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:629]: nit overall: Not a fan of having platform versions in job names or display names

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
I reran failed jobs. 🤞 that the macOS-10.15 failures are simply flakiness.

**Comment by wtgodbe** [comment]
If this fails again I'll try MacOs 11 - I only chose 10.15 here because 3.1 is "old"

**Comment by wtgodbe** [comment]
`Microsoft.DotNet.Watcher.Tools.FunctionalTests.FileWatcherTests.MoveFile(usePolling: False)` now fails consistently on MacOS 10.15

> System.TimeoutException : The operation at /_/src/Tools/dotnet-watch/test/FileWatcherTests.cs:132 timed out after reaching the limit of 60000ms.

>    at Microsoft.AspNetCore.Testing.TaskExtensions.TimeoutAfter[T](Task`1 task, TimeSpan timeout, String filePath, Int32 lineNumber)
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.FileWatcherTests.<>c__DisplayClass6_0.<<MoveFile>b__0>d.MoveNext() in /_/src/Tools/dotnet-watch/test/FileWatcherTests.cs:line 133
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.FileWatcherTests.UsingTempDirectory(Func`2 func) in /_/src/Tools/dotnet-watch/test/FileWatcherTests.cs:line 411
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.FileWatcherTests.MoveFile(Boolean usePolling) in /_/src/Tools/dotnet-watch/test/FileWatcherTests.cs:line 101
--- End of stack trace from previous location where exception was thrown ---

**Comment by wtgodbe** [comment]
@pranavkm should we just skip this dotnet-watch test on MacOs?

**Comment by pranavkm** [comment]
Yup, I'm fine with that.

**Pull request merged**